src/adapters/cloudflare/mod.rs

```diff
@@ -65,8 +66,104 @@ impl CloudFlareClient {
     }
 
     // For the monitor to grab metrics within a time range.
-    pub fn collect_metrics(&self) -> Result<()> {
-        todo!();
+    pub async fn collect_metrics(
+        &self,
+        account_id: String,
+        worker_name: String,
+        worker_version_id: String,
+        status_code_range_start: u16,
+        status_code_range_end: u16,
+        from_time: chrono::DateTime<chrono::Utc>,
+        to_time: chrono::DateTime<chrono::Utc>,
```

Comment:
This is way too many arguments for a function, but until we enable linting you're off the hook.